packages/pluggableWidgets/badge-web/src/components/Badge.tsx

```diff
@@ -1,30 +1,44 @@
-import { createElement, ReactElement } from "react";
+import { createElement, CSSProperties, KeyboardEvent, ReactElement, useCallback, useMemo } from "react";
 import classNames from "classnames";
 
-import "../ui/Badge.css";
-import { BrandStyleEnum } from "../../typings/BadgeProps";
-
 export interface BadgeProps {
     type: "badge" | "label";
     className?: string;
-    style?: object;
+    style?: CSSProperties;
     value: string;
-    brandStyle?: BrandStyleEnum;
-    clickable?: boolean;
     onClick?: () => void;
+    tabIndex?: number;
 }
 
 export const Badge = (props: BadgeProps): ReactElement => {
-    const { type, className, style, value, brandStyle, clickable, onClick } = props;
+    const { type, className, style, value, onClick, tabIndex } = props;
+
+    const onKeyDown = useCallback(
+        (event: KeyboardEvent<HTMLSpanElement>) => {
+            if (onClick && (event.key === "Enter" || event.key === " ")) {
+                onClick();
+            }
+        },
+        [onClick]
+    );
+
+    const actualTabIndex = useMemo(() => {
```

Comment:
Move this logic to the parent component.. makes more sense to pass a correct tabIndex then changing it in the display component

Reply:
I think the display component is responsible to ensure it's not clickable (giving the `span` no tabindex). Otherwise, the props being passed to this display component can be inconsistent (passing no onclick, but giving it a tabindex). This is more robust code.

Reply:
`tabIndex` in this case will be a fixed value, unless you re-run the project in SP it would never change as it cannot be dynamic. Does it worth to add a memoization here as you already have it in the parent?

[Small article about](https://blog.logrocket.com/rethinking-hooks-memoization/)

Reply:
IIUC, the "actual tab index" is determined based on the presence of an onClick handler. This logic could be moved to the parent and passed to the "display badge" as a prop, clarifying the display component's responsibility as display-only. Becuase the value of the memo is a primitive, I think the memo is not needed. If the value is passed to the "display badge" as a primitive prop, it will only cause a re-render if the value is different.

If the intent is indeed "logic/domain component" and "display component" separation, I'd go further and pull the `onKeyDown` logic higher, too.

Reply:
The `useMemo` is indeed not necessary here. The tabIndex will remain the same during execution of the MX app and the logic to determine the actual tab index isn't heavy at all. Therefore, I removed it.

I also moved both the `actualTabIndex` and `onKeyDown` logic to the parent component. It's indeed more logical to have it there :)